qatrack/qa/admin.py

```
@@ -903,7 +906,13 @@ def clean(self):
 
         test_type = cleaned_data.get("type")
         user_changing_type = self.instance.type != test_type
-        has_history = models.TestInstance.objects.filter(unit_test_info__test=self.instance).exists()
+        try:
+            has_history = models.TestInstance.objects.filter(unit_test_info__test=self.instance).exists()
+            print("has history: %s" % has_history, self.instance)
```

Ruff (0.15.14)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)